src/control/createIndex.ts

```diff
@@ -98,52 +85,134 @@ export interface CreateIndexPodSpec {
+    }
+    if (options.spec.serverless) {
+      ValidateProperties(options.spec.serverless, ['cloud', 'region']);
+      if (!options.spec.serverless.cloud) {
+        throw new PineconeArgumentError(
+          'You must pass a `cloud` for the serverless `spec` object in order to create an index.'
+        );
+      }
+      if (!options.spec.serverless.region) {
+        throw new PineconeArgumentError(
+          'You must pass a `region` for the serverless `spec` object in order to create an index.'
+        );
+      }
+    }
+    if (options.spec.pod) {
+      ValidateProperties(options.spec.pod, [
+        'environment',
+        'podType',
+        'replicas',
+        'shards',
+        'pods',
+        'metadataConfig',
+        'sourceCollection',
+      ]);
+      if (!options.spec.pod.environment) {
+        throw new PineconeArgumentError(
+          'You must pass an `environment` for the pod `spec` object in order to create an index.'
+        );
+      }
+      if (!options.spec.pod.podType) {
+        throw new PineconeArgumentError(
+          'You must pass a `podType` for the pod `spec` object in order to create an index.'
+        );
+      }
+    }
+    if (
+      options.spec.serverless &&
+      options.spec.serverless.cloud &&
+      !Object.values(ServerlessSpecCloudEnum).includes(
+        options.spec.serverless.cloud
+      )
+    ) {
+      throw new PineconeArgumentError(
+        `Invalid cloud value: ${
+          options.spec.serverless.cloud
+        }. Valid values are: ${Object.values(ServerlessSpecCloudEnum).join(
+          ', '
+        )}.`
+      );
+    }
+    if (
+      options.metric &&
+      !Object.values(IndexModelMetricEnum).includes(options.metric)
+    ) {
+      {
+        throw new PineconeArgumentError(
+          `Invalid metric value: ${options.metric}. Valid values are: 'cosine', 'euclidean', or 'dotproduct.'`
+        );
+      }
+    }
+    if (
+      options.spec.pod &&
+      options.spec.pod.replicas &&
+      options.spec.pod.replicas <= 0
+    ) {
+      throw new PineconeArgumentError(
+        'You must pass a positive integer for `replicas` in order to create an index.'
+      );
+    }
+    if (
+      options.spec.pod &&
+      options.spec.pod.pods &&
+      options.spec.pod.pods <= 0
+    ) {
+      throw new PineconeArgumentError(
+        'You must pass a positive integer for `pods` in order to create an index.'
+      );
+    }
+    if (options.spec.pod && !ValidPodTypes.includes(options.spec.pod.podType)) {
+      throw new PineconeArgumentError(
+        `Invalid pod type: ${
+          options.spec.pod.podType
+        }. Valid values are: ${ValidPodTypes.join(', ')}.`
+      );
+    }
+  };
 
   return async (options: CreateIndexOptions): Promise<IndexModel | void> => {
     // If metric is not specified, default to cosine
     if (options && !options.metric) {
-      options.metric = 'cosine';
+      options.metric = IndexModelMetricEnum.Cosine;
```

Comment:
Small nit improvement to use the `Enum` instead of hard-coding the `'cosine'` val here.